core/unix/src/TUnixSystem.cxx

```diff
@@ -568,6 +573,94 @@ TUnixSystem::~TUnixSystem()
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Initialize Unix system interface.
+extern "C"
+{
+   static int FullWrite(int fd, const char *text)
```

Comment:
I referred to Brian's code in cmssw. I guess it is called "FullWrite" is because it has been used in signal handler and we want to make sure it is async-signal-safe (not interrupted in the middle).